luxonis_train/config/config.py

```diff
@@ -82,6 +83,44 @@ class ModelConfig(BaseModelExtraForbid):
     visualizers: list[AttachedModuleConfig] = []
     outputs: list[str] = []
 
+    @field_validator("nodes", mode="before")
+    @classmethod
+    def validate_nodes(cls, nodes: Any) -> Any:
+        logged_general_warning = False
+        if not isinstance(nodes, list):
+            return nodes
+        names = []
+        last_body_index: int | None = None
+        for i, node in enumerate(nodes):
+            name = node.get("alias", node.get("name"))
+            if name is None:
+                raise ValueError(
+                    f"Node {i} does not specify the `name` field."
+                )
+            if "Head" in name and last_body_index is None:
```

Comment:
This requires "Head" to be part of the class name to automatically work right?

Reply:
Yeah, more specifically when first head is encountered, it assumes the node before was a neck and all subsequent nodes will be connected to it. By default, it just connects it to the previous one. So the `"Head"` in name is only required if you have multiple heads. For simply linear networks it will work no matter the names.